pkg/backup/backup.go

```diff
@@ -257,30 +262,147 @@ func (kb *kubernetesBackupper) Backup(log logrus.FieldLogger, backupRequest *Req
 		}
 	}
 
-	gb := kb.groupBackupperFactory.newGroupBackupper(
-		log,
-		backupRequest,
-		kb.dynamicFactory,
-		kb.discoveryHelper,
-		cohabitatingResources(),
-		kb.podCommandExecutor,
-		tw,
-		resticBackupper,
-		newPVCSnapshotTracker(),
-		volumeSnapshotterGetter,
-	)
+	// set up a temp dir for the itemCollector to use to temporarily
+	// store items as they're scraped from the API.
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return errors.Wrap(err, "error creating temp dir for backup")
+	}
+	defer os.RemoveAll(tempDir)
+
+	collector := &itemCollector{
+		log:                   log,
+		backupRequest:         backupRequest,
+		discoveryHelper:       kb.discoveryHelper,
+		dynamicFactory:        kb.dynamicFactory,
+		cohabitatingResources: cohabitatingResources(),
+		dir:                   tempDir,
+	}
+
+	items := collector.getAllItems()
+	log.WithField("progress", "").Infof("Collected %d resources from the Kubernetes API to back up", len(items))
+
+	itemBackupper := &defaultItemBackupper{
+		backupRequest:           backupRequest,
+		tarWriter:               tw,
+		dynamicFactory:          kb.dynamicFactory,
+		discoveryHelper:         kb.discoveryHelper,
+		resticBackupper:         resticBackupper,
+		resticSnapshotTracker:   newPVCSnapshotTracker(),
+		volumeSnapshotterGetter: volumeSnapshotterGetter,
+		itemHookHandler: &defaultItemHookHandler{
+			podCommandExecutor: kb.podCommandExecutor,
+		},
+	}
+	itemBackupper.additionalItemBackupper = itemBackupper
```

Comment:
TODO we should be able to get rid of the `additionalItemBackupper` field, since it was there only for testing purposes and isn't needed anymore.